reframe/core/variables.py

```diff
@@ -34,27 +34,47 @@ class TestVar:
     during class instantiation.
 
     To support injecting attributes into the variable, this class implements a
-    separate dict `__attrs__` where these will be stored.
+    separate dict `_attrs` where these will be stored.
 
     :meta private:
     '''
 
     __slots__ = (
-        'field', '_default_value', 'name', '__attrs__'
+        '_attrs', '_conv_fn', '_default_value', 'field', 'name'
     )
 
     def __init__(self, *args, **kwargs):
         field_type = kwargs.pop('field', fields.TypedField)
         self._default_value = kwargs.pop('value', Undefined)
-
         if not issubclass(field_type, fields.Field):
             raise TypeError(
                 f'field {field_type!r} is not derived from '
                 f'{fields.Field.__qualname__}'
             )
 
+        conv_fn = kwargs.pop('conv', None)
```

Comment:
If we leave the conversion to the descriptors, there is no need to add a conversion function on every variable declaration, which is kind of inconvenient.